### PR TITLE
cukes-sql: new module for DB testing

### PR DESCRIPTION
**Pull request by viktor-yengovatov**

### CODE REVIEW

File: cukes-sql/src/main/java/lv/ctco/cukes/sql/GenericTableRepository.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @SneakyThrows(SQLException.class)
+    public Integer countTableValues(String schema, String tableName) {
+        String query = createCountQuery(schema, tableName);
+        return Integer.valueOf(tryExecuteQuery(Collections.singletonList(ROW_COUNT_COLUMN), query).get(0).get(ROW_COUNT_COLUMN));
+    }
+
+	private List<Map<String, String>> tryExecuteQuery(List<String> columns, String query) throws SQLException {
+        try (Connection connection = connectionFactory.getConnection();
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(query)) {
```
Comment (sergeytrasko) [inline cukes-sql/src/main/java/lv/ctco/cukes/sql/GenericTableRepository.java:41]: ResultSet should be closed as well

File: cukes-sql/src/main/java/lv/ctco/cukes/sql/interceptor/InflateContextInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.aopalliance.intercept.MethodInvocation;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class InflateContextInterceptor implements MethodInterceptor {
```
Comment (sergeytrasko) [inline cukes-sql/src/main/java/lv/ctco/cukes/sql/interceptor/InflateContextInterceptor.java]: What's the difference to lv.ctco.cukes.core.internal.context.InflateContextInterceptor ?

File: cukes-sql/src/main/java/lv/ctco/cukes/sql/utils/TableAssertUtils.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        assertWithSign(assertThat(actual).as("Table row count should be %s %s", sign, number), sign, number);
+	}
+
+	private static void assertWithSign(AbstractIntegerAssert<?> anAssert, String sign, Integer number) {
+        switch (sign) {
+            case "<":
+                anAssert.isLessThan(number); break;
+            case "<=":
+                anAssert.isLessThanOrEqualTo(number); break;
+            case "!=":
```
Comment (sergeytrasko) [inline cukes-sql/src/main/java/lv/ctco/cukes/sql/utils/TableAssertUtils.java:34]: Please add '<>' as alternative here

File: cukes-sql/src/test/resources/cukes.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+cukes.plugins: lv.ctco.cukes.sql.CucumberBootstrap
+cukes.ltpa_enabled: false
```
Comment (sergeytrasko) [inline cukes-sql/src/test/resources/cukes.properties]: Not needed

File: cukes-sql/src/test/resources/cukes.properties [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+cukes.url_encoding_enabled: true
```
Comment (sergeytrasko) [inline cukes-sql/src/test/resources/cukes.properties]: Not needed

File: cukes-sql/pom.xml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>cukes-sql</artifactId>
+    <name>Cukes SQL</name>
+
+    <parent>
+        <groupId>lv.ctco.cukes</groupId>
+        <artifactId>cukes</artifactId>
+        <version>0.0.22-SNAPSHOT</version>
```
Comment (sergeytrasko) [inline cukes-sql/pom.xml:14]: Probably version should be adjusted before merging

**Review by sergeytrasko — CHANGES_REQUESTED** [review]
please address findings. Especially - close resources

File: cukes-sql/pom.xml (hunk 6 above)
Comment (viktor-yengovatov) [reply cukes-sql/pom.xml:14]: updated from master

File: cukes-sql/src/main/java/lv/ctco/cukes/sql/GenericTableRepository.java (hunk 1 above)
Comment (viktor-yengovatov) [reply cukes-sql/src/main/java/lv/ctco/cukes/sql/GenericTableRepository.java:41]: all fine here: connection, statement and resultSet are closed in Try-with-resources

**Review by sergeytrasko — APPROVED** [review]

**Pull request merged**